interoperability_report.py

```diff
@@ -0,0 +1,655 @@
+#!/usr/bin/python
+
+from multiprocessing import Process, Queue
+import time
+import re
+import pexpect
+import multiprocessing
+import argparse 
+import os
+from junitparser import TestCase, TestSuite, JUnitXml, Error, Attr
+from datetime import datetime
+
+from utilities import ReturnCode, path_executables
+from testSuite import dict_param_expected_code_timeout
+
+
+def subscriber(name_executable, parameters, key, time_out, code, data,
+                subscriber_finished, publisher_finished):
+    """ Run the executable with the parameters and save 
+        the error code obtained
+
+        name_executable     : name of the ShapeApplication to run
+                              as a Subscriber
+        parameters          : QOS to use with the Shape Application
+        key                 : test is being tested 
+                             (from dict_param_expected_code_timeout)
+        time_out            : time pexpect waits until it finds a pattern
+        code                : this variable will be overwritten with 
+                              the obtained ReturnCode
+        data                : this variable contains the samples 
+                              the Publisher sends
+        subscriber_finished : object event from multiprocessing that is set
+                              when the subscriber is finished
+        publisher_finished  : object event from multiprocessing that is set
+                              when the publisher is finished
+
+        The function runs the Shape Application as a Subscriber 
+        with the QoS defined.
+        It follows the next steps: 
+            * Wait until the topic is created
+            * Wait until the reader is created
+            * Wait until the reader matches with a writer
+            * Wait until the reader detects the writer as alive
+            * Wait until the reader receives data
+    
+    """
+    
+    # Step 1 : run the executable
+    child_sub = pexpect.spawnu(f'{name_executable} {parameters}')
+
+    # Save the child's output in the file created
+    msg_sub_verbose = open('log_sub.txt', 'w') 
+    child_sub.logfile = msg_sub_verbose
+   
+    # Step 2 : Check if the topic is created
+    index = child_sub.expect(
+        [
+            'Create topic:', 
```

Comment:
add the numbers that we use in the if as comments so it is easier to match them when reading the code
```suggestion
            'Create topic:', # index == 1
```

Reply:
okey